### PR TITLE
FEC-11453 HLS Widevine Offline Support

### PR DESCRIPTION
**Pull request by GouravSna**

### CODE REVIEW

File: tvplayer/src/main/java/com/kaltura/tvplayer/offline/exo/ExoOfflineManager.java [hunk 1]
```diff
@@ -1164,32 +1173,68 @@ private String extractDrmLicense(List<PKDrmParams> drmData, PKDrmParams.Scheme s
...
+            case dash: {
+                final DashManifest dashManifest = DashUtil.loadManifest(cacheDataSource, Uri.parse(contentUri));
+                Format formatWithDrmInitData = DashUtil.loadFormatWithDrmInitData(cacheDataSource, dashManifest.getPeriod(0));
+                drmInitData = formatWithDrmInitData == null ? null : formatWithDrmInitData.drmInitData;
+                log.d("Loading Dash drmInitData from remote");
+                break;
+            }
+            case hls: {
+                HlsPlaylist hlsPlaylist = ParsingLoadable.load(cacheDataSource, new HlsPlaylistParser(), Uri.parse(contentUri), C.DATA_TYPE_MANIFEST);
+                HlsMasterPlaylist hlsMasterPlaylist = (HlsMasterPlaylist) hlsPlaylist;
```
Comment (GouravSna) [inline tvplayer/src/main/java/com/kaltura/tvplayer/offline/exo/ExoOfflineManager.java]: Check the instance of HlsMasterPlaylist

File: tvplayer/src/main/java/com/kaltura/tvplayer/offline/exo/ExoOfflineManager.java [hunk 2]
```diff
@@ -1164,32 +1173,68 @@ private String extractDrmLicense(List<PKDrmParams> drmData, PKDrmParams.Scheme s
...
+                if (!hlsMasterPlaylist.variants.isEmpty()) {
+                    HlsMediaPlaylist hlsMediaPlaylist = (HlsMediaPlaylist) ParsingLoadable.load(cacheDataSource, new HlsPlaylistParser(), Uri.parse(hlsMasterPlaylist.variants.get(0).url.toString()), C.DATA_TYPE_MANIFEST);
```
Comment (GouravSna) [inline tvplayer/src/main/java/com/kaltura/tvplayer/offline/exo/ExoOfflineManager.java]: Check the instance for HlsMediaPlaylist as well

File: tvplayer/src/main/java/com/kaltura/tvplayer/offline/dtg/DTGOfflineManager.java [hunk 3]
```diff
@@ -9,6 +9,7 @@
 
 import com.kaltura.android.exoplayer2.database.DatabaseProvider;
 import com.kaltura.android.exoplayer2.upstream.cache.Cache;
+import com.kaltura.dtg.AssetFormat;
```
Comment (giladna) [inline tvplayer/src/main/java/com/kaltura/tvplayer/offline/dtg/DTGOfflineManager.java:12]: @GouravSna  is it needed?

File: tvplayer/src/main/java/com/kaltura/tvplayer/offline/exo/ExoOfflineManager.java [hunk 4]
```diff
@@ -471,7 +477,7 @@ public void prepareAsset(@NonNull PKMediaEntry mediaEntry, @NonNull SelectionPre
         }
 
         MediaItem mediaItem = builder.build();
-        if (mediaFormat != PKMediaFormat.dash) {
+        if (mediaFormat != PKMediaFormat.dash || mediaFormat != PKMediaFormat.hls) {
```
Comment (giladna) [inline tvplayer/src/main/java/com/kaltura/tvplayer/offline/exo/ExoOfflineManager.java]: @GouravSna  sounds like it has to be && and not ||

File: tvplayer/src/main/java/com/kaltura/tvplayer/offline/exo/ExoOfflineManager.java (hunk 1 above)
Comment (GouravSna) [reply tvplayer/src/main/java/com/kaltura/tvplayer/offline/exo/ExoOfflineManager.java]: Done

File: tvplayer/src/main/java/com/kaltura/tvplayer/offline/exo/ExoOfflineManager.java (hunk 4 above)
Comment (GouravSna) [reply tvplayer/src/main/java/com/kaltura/tvplayer/offline/exo/ExoOfflineManager.java]: Done

File: tvplayer/src/main/java/com/kaltura/tvplayer/offline/exo/ExoOfflineManager.java (hunk 2 above)
Comment (GouravSna) [reply tvplayer/src/main/java/com/kaltura/tvplayer/offline/exo/ExoOfflineManager.java]: Done

**Review by giladna — APPROVED** [review]

**Pull request merged**